lib/OpenQA/Worker/Cache/Client.pm

```diff
@@ -0,0 +1,279 @@
+# Copyright (C) 2018 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+
+package OpenQA::Worker::Cache::Client;
+
+use OpenQA::Worker::Cache;
+use OpenQA::Worker::Common;
+
+use OpenQA::Worker::Cache::Request;
+use OpenQA::Worker::Cache::Request::Asset;
+use OpenQA::Worker::Cache::Request::Sync;
+
+use Mojo::Base 'Mojo::UserAgent';
```

Comment:
Regarding the design of this class, are there good reasons for why it should be a subclass of `Mojo::UserAgent`? Otherwise maybe a `has a` relationship might be preferable to `is a`. If you're not planning on overloading anything the user agent could just be included with the delegation pattern as an attribute.
```perl
use Mojo::Base -base;
...
has ua => sub { Mojo::UserAgent->new };
...
$self->ua->post(...);
```

Reply:
I don't know - I've always thought at this as the Client ```is a``` specific UserAgent, instead that Client ```has a``` UserAgent, which also spares extra typing :smiley: of ```$self->ua```

Reply:
Yea, i can see your point. From my point of view subclassing has a much higher risk of possible breakage. Like if `Mojo::UserAgent` also started having a `host` attribute. So my instinct is to be more defensive here and always prefer delegation. 😉

Reply:
Addressed in https://github.com/os-autoinst/openQA/pull/1783/commits/9ecd435fceab4b1f57eca1088320dcf1a1628718